util/git/creds.go

```diff
@@ -673,3 +679,61 @@ func (c GoogleCloudCreds) getAccessToken() (string, error) {
 
 	return token.AccessToken, nil
 }
+
+type AzureWorkloadIdentityCreds struct {
+	store         CredsStore
+	tokenProvider workloadidentity.TokenProvider
+}
+
+func NewAzureWorkloadIdentityCreds(store CredsStore, tokenProvider workloadidentity.TokenProvider) AzureWorkloadIdentityCreds {
+	return AzureWorkloadIdentityCreds{
+		store:         store,
+		tokenProvider: tokenProvider,
+	}
+}
+
+// GetUserInfo returns the username and email address for the credentials, if they're available.
+func (c AzureWorkloadIdentityCreds) GetUserInfo(ctx context.Context) (string, string, error) {
+	// Email not implemented for HTTPS creds.
+	return "00000000-0000-0000-0000-000000000000", "", nil
```

Comment:
Should we return a non-nil error?

Reply:
Did not get the reason to return a non-nil error?

Reply:
Idk, was mostly just curious